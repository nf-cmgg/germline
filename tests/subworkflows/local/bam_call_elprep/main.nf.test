nextflow_workflow {

    name "Test Workflow BAM_CALL_ELPREP"
    script "subworkflows/local/bam_call_elprep/main.nf"
    workflow "BAM_CALL_ELPREP"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_call_elprep"
    tag "vcf_dbsnp_vcfanno"

    test("bam_call_elprep - default") {


        when {
            params {
                callers = "elprep"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143.00001", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split1, checkIfExists:true)
                ],[
                    [id:"NA24143.00002", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split2, checkIfExists:true)
                ],[
                    [id:"NA24143.00003", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split3, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.elfasta, checkIfExists:true)
                ])
                input[2] = [[],[]]
                input[3] = [[],[]]
                input[4] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.gvcfs.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] },
                    workflow.out.reports
                ).match() }
            )
        }

    }

    test("bam_call_elprep - dbsnp") {


        when {
            params {
                callers = "elprep"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143.00001", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split1, checkIfExists:true)
                ],[
                    [id:"NA24143.00002", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split2, checkIfExists:true)
                ],[
                    [id:"NA24143.00003", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", split_count:3],
                    file(params.bam1, checkIfExists:true),
                    file(params.bai1, checkIfExists:true),
                    file(params.split3, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.elfasta, checkIfExists:true)
                ])
                input[2] = [[],[]]
                input[3] = [[id:'dbsnp'], file(params.vcf1, checkIfExists:true)]
                input[4] = [[id:'dbsnp'], file(params.tbi1, checkIfExists:true)]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.gvcfs.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] },
                    workflow.out.reports
                ).match() }
            )
        }

    }
}
