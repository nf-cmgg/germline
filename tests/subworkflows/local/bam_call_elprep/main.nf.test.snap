{
    "bam_call_elprep - dbsnp": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "caller": "elprep"
                    },
                    "NA24143.elprep.g.vcf.gz,variantsMD5:974ed65cfad6264db7c6589d6b7d7d74",
                    "NA24143.elprep.g.vcf.gz.tbi"
                ]
            ],
            [
                [
                    "NA24143.elprep.bcftools_stats.txt:md5,36b9f979c03b24d87e2dc710baf3672b"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-04T15:56:30.054330853"
    },
    "bam_call_elprep - default": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "caller": "elprep"
                    },
                    "NA24143.elprep.g.vcf.gz,variantsMD5:974ed65cfad6264db7c6589d6b7d7d74",
                    "NA24143.elprep.g.vcf.gz.tbi"
                ]
            ],
            [
                [
                    "NA24143.elprep.bcftools_stats.txt:md5,36b9f979c03b24d87e2dc710baf3672b"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-04T15:55:56.561058236"
    }
}